<a href="https://colab.research.google.com/github/zhiyuguo-innis/MSSP607/blob/main/zhiyuguo-innis/MSSP607/Modules/Week_4/PARTICIPATION_ACTIVITY_Week_4_Opening_a_CSV_file_and_modifying_the_contents.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# PARTICIPATION ACTIVITY Week 4: Opening a CSV file and modifying the contents
**MSSP 607** · Zhiyu Guo

Data: `parsed.csv` (USGS earthquake data). Source: Molin, S. (2019). *Hands-On Data Analysis with Pandas*, Ch. 2.

## Setup
The notebook and data are stored together in my MSSP607 GitHub repository:
```
MSSP607/
└── Modules/
    └── Week_4/
        ├── PARTICIPATION_ACTIVITY_Week_4_...ipynb
        └── parsed.csv
```
The data is read directly from GitHub, so the notebook runs in Colab without uploading any files.

In [1]:
import pandas as pd

DATA_URL = (
    "https://raw.githubusercontent.com/zhiyuguo-innis/mssp607/main/"
    "zhiyuguo-innis/MSSP607/Modules/Week_4/parsed.csv"
)

df = pd.read_csv(DATA_URL)

print(df.shape)
df.head()

(9332, 27)


,alert,cdi,code,detail,dmin,felt,gap,ids,mag,magType,...,status,time,title,tsunami,type,types,tz,updated,url,parsed_place
0,NaN,NaN,37389218,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.008693,NaN,85.0,",ci37389218,",1.35,ml,...,automatic,1539475168010,"M 1.4 - 9km NE of Aguanga, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,",-480.0,1539475395144,https://earthquake.usgs.gov/earthquakes/eventp...,California
1,NaN,NaN,37389202,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.020030,NaN,79.0,",ci37389202,",1.29,ml,...,automatic,1539475129610,"M 1.3 - 9km NE of Aguanga, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,",-480.0,1539475253925,https://earthquake.usgs.gov/earthquakes/eventp...,California
2,NaN,4.4,37389194,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.021370,28.0,21.0,",ci37389194,",3.42,ml,...,automatic,1539475062610,"M 3.4 - 8km NE of Aguanga, CA",0,earthquake,",dyfi,focal-mechanism,geoserve,nearby-cities,o...",-480.0,1539536756176,https://earthquake.usgs.gov/earthquakes/eventp...,California
3,NaN,NaN,37389186,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.026180,NaN,39.0,",ci37389186,",0.44,ml,...,automatic,1539474978070,"M 0.4 - 9km NE of Aguanga, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,",-480.0,1539475196167,https://earthquake.usgs.gov/earthquakes/eventp...,California
4,NaN,NaN,73096941,https://earthquake.usgs.gov/fdsnws/event/1/que...,0.077990,NaN,192.0,",nc73096941,",2.16,md,...,automatic,1539474716050,"M 2.2 - 10km NW of Avenal, CA",0,earthquake,",geoserve,nearby-cities,origin,phase-data,scit...",-480.0,1539477547926,https://earthquake.usgs.gov/earthquakes/eventp...,California


## a. 95th percentile of earthquake magnitude in Japan (magType `mb`)

In [2]:
japan_mb = df[(df["parsed_place"] == "Japan") & (df["magType"] == "mb")]

print("Number of earthquakes:", len(japan_mb))
japan_mb["mag"].quantile(0.95)

Number of earthquakes: 50


np.float64(4.9)

**Answer:** Among the 50 `mb`-type earthquakes in Japan, the 95th percentile magnitude is **4.9**.

## b. Percentage of earthquakes in Indonesia coupled with tsunamis

In [3]:
indonesia = df[df["parsed_place"] == "Indonesia"]

indonesia_total = len(indonesia)
indonesia_tsunamis = indonesia["tsunami"].sum()
percentage = indonesia_tsunamis / indonesia_total * 100

print("Total earthquakes:", indonesia_total)
print("Earthquakes with tsunamis:", indonesia_tsunamis)
print("Percentage:", round(percentage, 2), "%")

Total earthquakes: 147
Earthquakes with tsunamis: 34
Percentage: 23.13 %


**Answer:** 34 of 147 earthquakes in Indonesia (**23.13%**) were coupled with tsunamis.

## c. Summary statistics for earthquakes in Nevada

In [4]:
nevada = df[df["parsed_place"] == "Nevada"]

nevada.describe()

,cdi,dmin,felt,gap,mag,mmi,nst,rms,sig,time,tsunami,tz,updated
count,15.000000,681.000000,15.000000,681.000000,681.000000,1.00,681.000000,681.000000,681.000000,6.810000e+02,681.0,681.0,6.810000e+02
mean,2.440000,0.166199,2.400000,153.668120,0.500073,2.84,12.618209,0.151986,10.970631,1.538314e+12,0.0,-480.0,1.538402e+12
std,0.501142,0.166228,4.626013,68.735302,0.696710,NaN,9.866963,0.084662,19.607150,5.965637e+08,0.0,0.0,6.010951e+08
min,2.000000,0.001000,1.000000,29.140000,-0.500000,2.84,3.000000,0.000500,0.000000,1.537247e+12,0.0,-480.0,1.537307e+12
25%,2.000000,0.053000,1.000000,97.380000,-0.100000,2.84,6.000000,0.106900,0.000000,1.537854e+12,0.0,-480.0,1.537928e+12
50%,2.200000,0.112000,1.000000,149.140000,0.400000,2.84,10.000000,0.146300,2.000000,1.538280e+12,0.0,-480.0,1.538428e+12
75%,2.900000,0.233000,1.000000,199.720000,0.900000,2.84,16.000000,0.187100,12.000000,1.538821e+12,0.0,-480.0,1.538878e+12
max,3.300000,1.414000,19.000000,355.910000,2.900000,2.84,61.000000,0.863400,129.000000,1.539461e+12,0.0,-480.0,1.539483e+12


In [5]:
nevada["mag"].describe()

,mag
count,681.000000
mean,0.500073
std,0.696710
min,-0.500000
25%,-0.100000
50%,0.400000
75%,0.900000
max,2.900000


**Interpretation:** There are **681** earthquakes in Nevada. They are very small: the mean magnitude is about **0.50**, the median is **0.4**, and the range is **-0.5 to 2.9**. None were associated with a tsunami, which is expected since Nevada is landlocked. Only 15 events have "felt" reports (`felt`, `cdi`), consistent with most quakes being too weak to notice.

## d. Ring of Fire indicator column
Notes on matching:
- **Exact matching** (`isin`) is used for most places, so `New Mexico` is **not** selected when matching `Mexico`.
- **Antarctica** does not appear as `"Antarctica"` in `parsed_place`. It shows up as ocean ridges (e.g. `Pacific-Antarctic Ridge`), so those are matched by the substring `"Antarctic"`.

In [6]:
# Check how the tricky place names appear in the data
df[df["parsed_place"].str.contains("Antarctic|Mexico|Kermadec", na=False)]["parsed_place"].value_counts()

,count
parsed_place,
Mexico,38
Pacific-Antarctic Ridge,3
Kermadec Islands,2
Western Indian-Antarctic Ridge,1
New Mexico,1


In [7]:
ring_of_fire_places = [
    "Bolivia", "Chile", "Ecuador", "Peru", "Costa Rica", "Guatemala", "Mexico",
    "Japan", "Philippines", "Indonesia", "New Zealand", "Canada", "Fiji",
    "Alaska", "Washington", "California", "Russia", "Taiwan", "Tonga",
    "Kermadec Islands"
]

df["ring_of_fire"] = (
    df["parsed_place"].isin(ring_of_fire_places)
    | df["parsed_place"].str.contains("Antarctic", na=False)
)

# Sanity checks
print("New Mexico flagged?", df.loc[df["parsed_place"] == "New Mexico", "ring_of_fire"].any())
print("Antarctic rows flagged:", df.loc[df["parsed_place"].str.contains("Antarctic", na=False), "ring_of_fire"].sum())

df[["parsed_place", "ring_of_fire"]].head(10)

New Mexico flagged? False
Antarctic rows flagged: 4


,parsed_place,ring_of_fire
0,California,True
1,California,True
2,California,True
3,California,True
4,California,True
5,Dominican Republic,False
6,Alaska,True
7,California,True
8,California,True
9,Indonesia,True


## e. Number of earthquakes inside vs. outside the Ring of Fire

In [8]:
df["ring_of_fire"].value_counts()

,count
ring_of_fire,
True,7188
False,2144


**Answer:** **7,188** earthquakes occurred in Ring of Fire locations and **2,144** occurred outside them.

## f. Tsunami count along the Ring of Fire

In [9]:
ring_of_fire_tsunamis = df[df["ring_of_fire"] & (df["tsunami"] == 1)]

len(ring_of_fire_tsunamis)

45

**Answer:** **45** earthquakes along the Ring of Fire were associated with tsunamis.